SAML2/XML/Schema/Datatypes.hs

```diff
@@ -78,13 +79,16 @@ xpDateTime :: XP.PU DateTime
 xpDateTime = XP.PU
   { XP.theSchema = XPS.scDTxsd XSD.xsd_dateTime []
   , XP.appPickle = XP.putCont . XN.mkText . tweakTimeString . formatTime defaultTimeLocale fmtz
-  , XP.appUnPickle = XP.getCont >>= XP.liftMaybe "dateTime expects text" . XN.getText >>= parseTimeM True defaultTimeLocale fmtz
+  , XP.appUnPickle = XP.getCont >>= XP.liftMaybe "dateTime expects text" . XN.getText >>= parseTime
```

Comment:
I have opened a PR against `hxt` to add a `MonadFail` instance for the `Unpickler` type (https://github.com/UweSchmidt/hxt/pull/85) which (if it gets merged) would probably be nicer than the explicit workaround here?